ui/src/main/webapp/app/query/operation-chain-component.js

```diff
@@ -135,6 +135,66 @@ function OperationChainController(operationChain, config, loading, query, error,
         });
     }
 
+    vm.saveChain = function(ev) {
+        events.broadcast('onPreExecute', []);
+        if (!vm.canExecute()) {
+            return;
+        }
+
+        if (vm.operations.length === 0) {
+            error.handle("Unable to save operation chain with no operations");
+            return;
+        }
+
+        var chain = {
+            class: OPERATION_CHAIN_CLASS,
+            operations: []
+        }
+        for (var i in vm.operations) {
+            chain.operations.push(createOperationForQuery(vm.operations[i]));
+        }
+
+        var confirm = $mdDialog.confirm()
+        .title('Operation chain saved as named operation!')
+        .textContent('You can now see your saved operation in the list of operations')
+        .targetEvent(ev)
+        .ok('Ok')
+
+        var invalidName = $mdDialog.confirm()
+        .title('Operation chain name is invalid!')
+        .textContent('The operation chain cannot be empty')
+        .targetEvent(ev)
+        .ok('Ok')
+
+        if (vm.operationName != null) {
+            query.executeQuery(
+                {
+                    class: ADD_NAMED_OPERATION_CLASS,
+                    operationName: vm.operationName,
+                    operationChain: chain,
+                    description: 'test description',
```

Comment:
Description should not be hard coded.

Reply:
Can now input the description in a textarea in the sidenav